arbor/backends/multicore/mechanism.cpp

```diff
@@ -163,22 +163,33 @@ void mechanism::instantiate(unsigned id, backend::shared_state& shared, const me
         copy_extend(pos_data.multiplicity, multiplicity_, 1);
     }
 
-    for (auto i: ion_index_table()) {
-        auto ion_binding = value_by_key(overrides.ion_rebind, i.first).value_or(i.first);
-
-        ion_state* oion = ptr_by_key(shared.ion_data, ion_binding);
-        if (!oion) {
-            throw arbor_internal_error("multicore/mechanism: mechanism holds ion with no corresponding shared state");
-        }
+    {
+        auto table = ion_index_table();
+        // Allocate bulk storage
+        auto count = (table.size() + 1)*width_padded_;
```

Comment:
Why `table.size()+1`? Shouldn't it be enough to allocate enough space for `table.size()`? In the loop below `table.size()` indexes are created, not `table.size()+1`.

Reply:
Yes.

Reply:
Done